apps/app/src/test/scala/org/lfdecentralizedtrust/splice/integration/tests/ScanHistoryBackfillingIntegrationTest.scala

```diff
@@ -387,6 +401,19 @@ class ScanHistoryBackfillingIntegrationTest
       commonLength should be > 10
       val sv1Items = sv1HttpUpdates.take(commonLength)
       val sv2Items = sv2HttpUpdates.take(commonLength)
+      def collectEventsById(items: Seq[UpdateHistoryItemV2]) = items.collect {
+        case definitions.UpdateHistoryItemV2.members.UpdateHistoryTransactionV2(http) =>
+          http.eventsById
+      }
+      val eventsByIdSv1 = collectEventsById(sv1Items)
+      val eventsByIdSv2 = collectEventsById(sv2Items)
+
+      def assertOrderedEventsById(eventsByIdSeq: Seq[SortedMap[String, definitions.TreeEvent]]) =
+        forAll(eventsByIdSeq) { eventsById =>
+          eventsById.keys.toSeq should contain theSameElementsInOrderAs SortedMap.from(eventsById).keys.toSeq
+        }
+      assertOrderedEventsById(eventsByIdSv1)
+      assertOrderedEventsById(eventsByIdSv2)
```

Comment:
How can these tests fail?

```suggestion
```

Reply:
yeah.. I initially had no x-scala-map so I guess it makes no sense anymore (nor did it at the time when I did not realize that the generated type was a map)